article-analyzer/src/main/java/com/reader/article_analyzer/ArticleCategoryAnalyzer.java

```diff
@@ -0,0 +1,68 @@
+package com.reader.article_analyzer;
+
+import com.reader.article_analyzer.Model.Article;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.*;
+
+@Component
+public class ArticleCategoryAnalyzer {
+    private final JsonFileService fileService;
+    private final String[] excludedWordsArray;
+    private final List<Set<String>> allCategories;
+
+    public ArticleCategoryAnalyzer(@Value("${excluded.words}") String excludedWords, JsonFileService fileService) {
+        this.excludedWordsArray = excludedWords.split(", ");
+        this.fileService = fileService;
+        this.allCategories = new ArrayList<>(); // Ініціалізуємо список категорій
+    }
+
+    public void analyzeArticleContent(String filePath, int articleId) throws IOException {
```

Comment:
this method should return results of all found categories from all articles

Reply:
```suggestion
    public List<Set<String>> analyzeArticleContent(String filePath, int articleId) throws IOException {
```